### PR TITLE
Remove useless code and reduce matching speed

### PR DESCRIPTION
**Pull request by victoria34**
1. optimize traverse_match_tree() to improve matching speed. 5 ~ 10 mins/round.
2. get oncotree from oncotree_mapping.json
3. process oncokb genomic criteria in prepare_oncokb_genomic_criteria()

Fixed https://github.com/oncokb/oncokb/issues/1894

### CODE REVIEW

File: matchengine.py [hunk 1]
```diff
@@ -104,15 +108,48 @@ def load_pkl(self, clinical, genomic):
...
+        subprocess.call(cmd2.split(' '))
+
+        # convert string to date object
+        for clinical_item in list(self.db.clinical.find()):
+            cols = list()
+            keys = clinical_item.keys()
+            if 'BIRTH_DATE' in keys:
+                cols.append('BIRTH_DATE')
+            if 'REPORT_DATE' in keys:
+                cols.append('REPORT_DATE')
```
Comment (zhx828) [inline matchengine.py]: Would it better that have a list of these special keys, for a for loop checking whether the key exists, if it does, append the key directly?

File: matchengine.py [hunk 2]
```diff
@@ -195,9 +232,9 @@ def load(args):
             db.clinical.insert(clinical_json)
 
             # Get clinical ids from mongo
-            logging.info('Adding clinical ids to genomic data...')
-            clinical_doc = list(db.clinical.find({}, {"_id": 1, "SAMPLE_ID": 1}))
-            clinical_dict = dict(zip([i['SAMPLE_ID'] for i in clinical_doc], [i['_id'] for i in clinical_doc]))
+            # logging.info('Adding clinical ids to genomic data...')
+            # clinical_doc = list(db.clinical.find({}, {"_id": 1, "SAMPLE_ID": 1}))
+            # clinical_dict = dict(zip([i['SAMPLE_ID'] for i in clinical_doc], [i['_id'] for i in clinical_doc]))
```
Comment (zhx828) [inline matchengine.py]: Is it safe to delete these comments?

File: matchengine.py [hunk 3]
```diff
@@ -299,6 +335,7 @@ def match(args):
     param_outpath_help = 'Destination and name of your results file.'
     param_trial_format_help = 'File format of input trial data. Default is YML.'
     param_patient_format_help = 'File format of input patient data (both clinical and genomic files). Default is CSV.'
+    param_match_method_help = 'Match method name. The default is oncokb and uses oncokb_match().'
```
Comment (zhx828) [inline matchengine.py]: Could you elaborate the parameter more? What the match for, and how that works?

File: matchengine/engine.py [hunk 4]
```diff
@@ -565,62 +818,78 @@ def _assess_match(self, mrn_map, trial_matches, trial, trial_segment, match_segm
...
-            clinical = list(self.db.clinical.find({'SAMPLE_ID': {'$in': list(sample_ids)}}, cproj))
+        # clinical = []
+        # if sample_ids:
+        #     cproj = {
+        #             'SAMPLE_ID': 1,
+        #             'VITAL_STATUS': 1,
+        #             'GENDER': 1,
+        #             '_id': 0
+        #         }
+        #     clinical = list(self.db.clinical.find({'SAMPLE_ID': {'$in': list(sample_ids)}}, cproj))
```
Comment (zhx828) [inline matchengine/engine.py]: Let's get rid of this?

**Review by zhx828 — APPROVED** [review]
Some minor comments, especially about commented out sections. If you think we should leave them there, please ignore my comments.

File: matchengine.py (hunk 2 above)
Comment (victoria34) [reply matchengine.py]: Yes, it's safe. I can delete them.

File: matchengine.py (hunk 3 above)
Comment (victoria34) [reply matchengine.py]: That's a draft language I wrote long time ago. I'll update it to make it cleaer. Thanks for pointing it out.

File: matchengine.py (hunk 1 above)
Comment (victoria34) [reply matchengine.py]: Good idea. We only need birthdate for now. I'll create list with birthday directly.

**Pull request merged**